nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/InputPortResource.java

```diff
@@ -315,6 +319,90 @@ public Response removeInputPort(
+                    @Authorization(value = "Write - /input-ports/{uuid} or /operation/input-ports/{uuid}")
+            }
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(code = 400, message = "NiFi was unable to complete the request because it was invalid. The request should not be retried without modification."),
+                    @ApiResponse(code = 401, message = "Client could not be authenticated."),
+                    @ApiResponse(code = 403, message = "Client is not authorized to make this request."),
+                    @ApiResponse(code = 404, message = "The specified resource could not be found."),
+                    @ApiResponse(code = 409, message = "The request was valid but NiFi was not in the appropriate state to process it. Retrying the same request later may be successful.")
+            }
+    )
+    public Response updateRunStatus(
+            @Context final HttpServletRequest httpServletRequest,
+            @ApiParam(
+                    value = "The port id.",
+                    required = true
+            )
+            @PathParam("id") final String id,
+            @ApiParam(
+                    value = "The port run status.",
+                    required = true
+            ) final PortRunStatusEntity requestRunStatus) {
+
+        if (requestRunStatus == null) {
+            throw new IllegalArgumentException("Port run status must be specified.");
+        }
+
+        if (requestRunStatus.getRevision() == null) {
+            throw new IllegalArgumentException("Revision must be specified.");
+        }
+
+        requestRunStatus.validateState();
+
+        if (isReplicateRequest()) {
+            return replicate(HttpMethod.PUT, requestRunStatus);
+        } else if (isDisconnectedFromCluster()) {
+            verifyDisconnectedNodeModification(requestRunStatus.isDisconnectedNodeAcknowledged());
+        }
+
+        // handle expects request (usually from the cluster manager)
+        final Revision requestRevision = getRevision(requestRunStatus.getRevision(), id);
+        // Create port DTO to verify if it can be updated.
+        final PortDTO portDTO = new PortDTO();
+        portDTO.setId(id);
+        portDTO.setState(requestRunStatus.getState());
+
+        return withWriteLock(
+                serviceFacade,
+                requestRunStatus,
+                requestRevision,
+                lookup -> {
+                    final NiFiUser user = NiFiUserUtils.getNiFiUser();
+
+                    final Authorizable authorizable = lookup.getInputPort(id);
+                    OperationAuthorizable.authorize(authorizable, authorizer, RequestAction.WRITE, user);
+                },
+                () -> serviceFacade.verifyUpdateInputPort(portDTO),
+                (revision, runStatusEntity) -> {
+                    // update the input port
+                    final PortEntity entity = serviceFacade.updateInputPort(revision, portDTO);
```

Comment:
We need to recreate this `portDTO` using the `runStatusEntity` due to how we authorize/cache requests during our two phase commit.

Reply:
Fixed.